src/GafferOSL/ShadingEngine.cpp

```diff
@@ -1121,16 +1191,16 @@ class ShadingResults
 // and the OSL machinery that needs to be stored per "renderer-thread"
 struct ThreadInfo
 {
-	ThreadInfo()
-		: oslThreadInfo( ::shadingSystem()->create_thread_info() ),
-		  shadingContext( ::shadingSystem()->get_context( oslThreadInfo ) )
+	ThreadInfo() :
+		oslThreadInfo( ::shadingSystem().first->create_thread_info() ),
+		shadingContext( ::shadingSystem().first->get_context( oslThreadInfo ) )
```

Comment:
Returning `std::pair` is making a bit of a mess of the majority of call sites, when only a single caller actually needs `.second`. I think a good old fashioned `int *width = nullptr` output argument would be superior here.

Reply:
Addressed in 8c84600